src/Symfony/Component/Process/Pipes/AbstractPipes.php

```diff
@@ -29,6 +29,17 @@
     /** @var bool */
     private $blocked = true;
 
+    public function __construct($input)
+    {
+        if (is_resource($input)) {
+            $this->input = $input;
+        } elseif (is_string($input)) {
+            $this->inputBuffer = $input;
+        } else {
+            $this->inputBuffer = (string) $input;
```

Comment:
Since the `(string)` casting operator duplicates string (at least on PHP 5), we use it only for non-strings.